components/tidb_query/src/codec/data_type/vector.rs

```diff
@@ -218,6 +219,47 @@ impl VectorValue {
         }
     }
 
+    /// Returns maximum encoded size in arrow format.
+    pub fn maximum_encoded_size_arrow(&self, logical_rows: &[usize]) -> Result<usize> {
```

Comment:
The calculation seems incorrect. When column length is very small (0 or 1), the size is smaller than actual. And we can also calculate a more tight upper bound.

Reply:
I add 10 bytes into the result, so it will be bigger than actual in the situation. It is ok?

Reply:
Yes, that's definitely enough.